autoray/lazy/core.py

```diff
@@ -1267,6 +1267,30 @@ def where(condition, x, y):
     )
 
 
+def indices_shape_for_lazy(indices):
+    import numpy as np
+    shape_list = list(np.array(indices).shape)
+    return shape_list
```

Comment:
Would a function like so suffice here? :
```python
def _get_py_shape(x):
    """Infer the shape of a possibly nested list/tuple object.
    """
    if hasattr(x, "shape"):
        return x.shape
    if isinstance(x, (tuple, list)):
        return (len(x),) + _get_py_shape(x[0])
    return ()
```
then we could avoid the `numpy` dependency.